📚 Docs: Clarify that `auth.uid()` returns `null` when unauthenticated in RLS policies

LakshRJain opened the pull request:
#37320 
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Docs update

## What is the current behavior?

The documentation does not explain what happens when `auth.uid()` is used in RLS policies without an active session. When `auth.uid()` evaluates to `null`, policies silently reject the query, leading to confusion for developers who assume their RLS or auth setup is broken.

There was no warning or explanation for this behavior in the existing docs.

## What is the new behavior?

A new warning has been added to the RLS documentation to clarify that:

- `auth.uid()` returns `null` if no session is present.
- Policies using `auth.uid()` (e.g., `auth.uid() = user_id`) will fail if not combined with an explicit `IS NOT NULL` check.

This helps developers understand and avoid unintended policy rejections.

## Additional context

This confusion is commonly encountered by new Supabase users. The HTTP request may look valid, but the RLS silently fails due to `null = user_id`.

This update improves developer experience and prevents hours of debugging.

LakshRJain committed:
📚 Fix: Clarify behavior of auth.uid() when unauthenticated in RLS

LakshRJain committed:
Update row-level-security.mdx

LakshRJain committed:
Update row-level-security.mdx

LakshRJain committed:
Update row-level-security.mdx

LakshRJain committed:
Merge branch 'master' into fix-auth-uid-null-doc

LakshRJain committed:
Merge branch 'master' into fix-auth-uid-null-doc

ChrisChinchilla commented on apps/docs/content/guides/database/postgres/row-level-security.mdx:
```diff
@@ -59,6 +59,26 @@ alter table "table_name" enable row level security;
 
 Once you have enabled RLS, no data will be accessible via the [API](/docs/guides/api) when using the public `anon` key, until you create policies.
 
+### ⚠️ Be Aware: `auth.uid()` Returns `null` When Unauthenticated
```
```suggestion
### ⚠️ Be Aware: `auth.uid()` Returns `null` When Unauthenticated
<Admonition type="warning" label="`auth.uid()` Returns `null` When Unauthenticated">
```

ChrisChinchilla commented on apps/docs/content/guides/database/postgres/row-level-security.mdx:
````diff
@@ -59,6 +59,26 @@ alter table "table_name" enable row level security;
+```sql
+USING (auth.uid() IS NOT NULL AND auth.uid() = user_id)
+```
+
````
```suggestion
</Admonition>
```

ChrisChinchilla commented on apps/docs/content/guides/database/postgres/row-level-security.mdx:
```diff
@@ -59,6 +59,26 @@ alter table "table_name" enable row level security;
 
+### ⚠️ Be Aware: `auth.uid()` Returns `null` When Unauthenticated
+
+When a request is made without an authenticated user (e.g., no access token is provided or the session has expired), `auth.uid()` will return `null`.
```
```suggestion
When a request is made without an authenticated user (e.g., no access token is provided or the session has expired), `auth.uid()` returns `null`.
```

ChrisChinchilla reviewed (dismissed):
just some small formatting and grammar tweaks

ChrisChinchilla committed:
Update apps/docs/content/guides/database/postgres/row-level-security.mdx

ChrisChinchilla committed:
Update apps/docs/content/guides/database/postgres/row-level-security.mdx

ChrisChinchilla committed:
Update apps/docs/content/guides/database/postgres/row-level-security.mdx

ChrisChinchilla committed:
Merge branch 'master' into fix-auth-uid-null-doc

ChrisChinchilla committed:
Update apps/docs/content/guides/database/postgres/row-level-security.mdx

ChrisChinchilla committed:
Update warning about `auth.uid()` behavior

Clarified behavior of `auth.uid()` when unauthenticated and recommended explicit authentication checks.

ChrisChinchilla committed:
Change admonition type from warning to caution

ChrisChinchilla committed:
Update SQL condition for row-level security

Clarify row-level security SQL condition for unauthenticated users.
